### ISSUE TITLE
Add Tertiary Button

### ISSUE THREAD

**Issue by scurker** [dequelabs/cauldron#1193]
![tertiary button styles in their resting, hover, focus, active, and disabled states](https://github.com/user-attachments/assets/ac2fce1f-c6fa-482d-ad94-428d4b8d1dcc)

Tertiary buttons are an alternate style of button that presents a low priority action taking less visual prominence than primary or secondary buttons. The above image shows the tertiary button in its resting, hover, focus, active, and disabled state.

## Implementation

Add an additional variant of `tertiary` for the `Button` component.

### States

The hover, focus, and active states are visually identical to the states of the secondary variant. The main different is that the resting and disabled states do not have a visible background or border color for the button. The `thin` modifier should still be supported for the `tertiary` variant.

## Additional Notes

Dark mode's text is white to contrast with the background, all other state changes are applicable.

This is co-dependent on #1595 which will handle the associated text sizes changing.
